src/umbraco.cms/businesslogic/Packager/PackageInstance/CreatedPackage.cs

```diff
@@ -347,21 +347,12 @@ public void Publish()
                 _packageManifest.Save(manifestFileName);
                 _packageManifest = null;
 
-
-                //string packPath = Settings.PackagerRoot.Replace(System.IO.Path.DirectorySeparatorChar.ToString(), "/") + "/" + pack.Name.Replace(' ', '_') + "_" + pack.Version.Replace(' ', '_') + "." + Settings.PackageFileExtension;
-
-                // check if there's a packages directory below media
-                var packagesDirectory = SystemDirectories.Media + "/created-packages";
-                if (Directory.Exists(IOHelper.MapPath(packagesDirectory)) == false)
-                {
-                    Directory.CreateDirectory(IOHelper.MapPath(packagesDirectory));
-                }
-
-
+                var packagesDirectory = "created-packages";
                 var packPath = packagesDirectory + "/" + (pack.Name + "_" + pack.Version).Replace(' ', '_') + "." + Settings.PackageFileExtension;
-                utill.ZipPackage(localPath, IOHelper.MapPath(packPath));
+                var packContents = utill.ZipPackage(localPath);
+                UmbracoMediaFile.Save(packContents, packPath);

```

Comment:
Can we always assume the folder will be created by the FileSystemProvider? 
I'd personally prefer using the IFileSystem OpenFile method as this isn't really a media file

Reply:
In case of Azure Storage you can't create any folders so the FileSystemProvider should handle it. I choose to use UmbracoMediaFile because it saves the file in the Media folder.